### PR TITLE
Update character _preUpdate logic for v12

### PR DESCRIPTION
**Pull request by CarlosFdez**
Fixes the unnecessary cloning

### CODE REVIEW

File: src/module/actor/character/document.ts [hunk 1]
```diff
@@ -1841,10 +1841,14 @@ class CharacterPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e
...
-        if (changed.system.details?.level || changed.system.build?.attributes) {
+        const newLevel = changed.system.details?.level?.value ?? this.level;
+        const attributeChanged =
+            !!changed.system.build?.attributes &&
+            !R.isEmpty(fu.diffObject(this._source.system.build?.attributes ?? {}, changed.system.build.attributes));
+        if (newLevel !== this.level || attributeChanged) {
             const level = changed.system.details?.level;
-            if (typeof level?.value === "number") {
-                level.value = Math.clamp(Number(level.value) || 0, 0, 30) || 0;
+            if (level) {
```
Comment (In3luki) [inline src/module/actor/character/document.ts:1850]: This doesn't handle level 0

Comment (CarlosFdez) [reply src/module/actor/character/document.ts:1850]: good catch

Comment (CarlosFdez) [reply src/module/actor/character/document.ts:1850]: Wait no, it does, level is an object. level.value is what would be level 0.

Comment (In3luki) [reply src/module/actor/character/document.ts:1850]: oh true

**Pull request merged**